_data-prepper/pipelines/dlq.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: default
+title: Dead-letter queues 
+parent: Pipelines
+nav_order: 13
+---
+
+# Dead-letter queues
+
+Data Prepper pipelines support dead-letter queues (DLQ) to offload failed events and make them accessible for analysis.
+
+As of Data Prepper 2.3, one the `s3` source supports DLQs.
+
+## Configure a DLQ writer
+
+To configure a DLQ writer for the `s3` source, add the following to your pipeline.yaml's `sink` configuration:
+
+```yaml
+  sink:
+    opensearch:
+      dlq:
+        s3:
+          bucket: "my-dlq-bucket"
+          key_path_prefix: "dlq-files/"
+          region: "us-west-2"
+          sts_role_arn: "arn:aws:iam::123456789012:role/dlq-role"
+```
+
+The resulting DLQ file outs as a JSON array of DLQ objects. Any files written to the S3 DLQ contains the following name pattern:
+
+```
+dlq-v${version}-${pipelineName}-${pluginId}-${timestampIso8601}-${uniqueId}
+```
+
+- `version`: The Data Prepper version.
+- `pipelineName`: The pipeline name indicated in pipeline.yaml.
+- `pluginId`: The ID of the plugin associated with the DLQ event.
+
+## Configuration
+
+DLQ supports the following configuration options.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+bucket | Yes | String | The bucket name of the DLQ outputs failed records.
```

Comment:
output's?

Reply:
Switched to The bucket name where the DLQ outputs failed records.